### PR TITLE
jinja2: update templates to use django-compressor (bug 1899795)

### PR DESCRIPTION
**Pull request by zzzeid**

### CODE REVIEW

File: src/lando/ui/jinja2/partials/layout.html [hunk 1]
```diff
@@ -5,9 +5,41 @@
...
+      <link type="text/x-scss" href="/static/legacy/css/components/Badge.scss" rel="stylesheet" media="screen">
+      <link type="text/x-scss" href="/static/legacy/css/components/FlashMessages.scss" rel="stylesheet" media="screen">
+      <link type="text/x-scss" href="/static/legacy/css/components/Footer.scss" rel="stylesheet" media="screen">
+
+      <link type="text/x-scss" href="/static/legacy/css/pages/StackPage.scss" rel="stylesheet" media="screen">
+      <link type="text/x-scss" href="/static/legacy/css/pages/RevisionPage.scss" rel="stylesheet" media="screen">
+      <link type="text/x-scss" href="/static/legacy/css/pages/TreestatusPage.scss" rel="stylesheet" media="screen">
+  {% endcompress %}
+
+  {% compress js file "vendor" %} {# "vendor" js files #}
```
Comment (zzzeid) [inline src/lando/ui/jinja2/partials/layout.html:28]:
```suggestion
  {% compress js file vendor %}
```

File: src/lando/ui/jinja2/partials/layout.html [hunk 2]
```diff
@@ -5,9 +5,41 @@
...
+      <script src="/static/legacy/vendor/jquery-3.2.1.min.js" type="text/javascript" charset="utf-8"></script>
+  {% endcompress %}
+
+  {% compress js file "main" %} {# "main" js files #}
```
Comment (zzzeid) [inline src/lando/ui/jinja2/partials/layout.html:32]:
```suggestion
  {% compress js file main %}
```

File: src/lando/ui/jinja2/partials/layout.html [hunk 3]
```diff
@@ -5,9 +5,41 @@
...
   <meta content="width=device-width,initial-scale=1" name="viewport">
-  <link rel="shortcut icon"
-        href="{{ url_for('static', filename='images/logo/bird_64.png') }}">
+  <link rel="shortcut icon" href="/static/assets/images/logo/bird_64.png">
   <link rel="stylesheet" href="https://code.cdn.mozilla.net/fonts/fira.css">
+
+  {% compress css file vendor %} {# "vendor" css files #}
+      <link type="text/css" href="/static/legacy/vendor/font-awesome.min.css" rel="stylesheet" media="screen">
+  {% endcompress %}
+  {% compress css file main %} {# "main" css files #}
```
Comment (zzzeid) [inline src/lando/ui/jinja2/partials/layout.html:14]:
```suggestion
  {% compress css file main %}
```

File: src/lando/ui/jinja2/partials/layout.html [hunk 4]
```diff
@@ -5,9 +5,41 @@
   <meta content="ie=edge" http-equiv="x-ua-compatible">
   <title>{% block page_title %}{% endblock %}</title>
   <meta content="width=device-width,initial-scale=1" name="viewport">
-  <link rel="shortcut icon"
-        href="{{ url_for('static', filename='images/logo/bird_64.png') }}">
+  <link rel="shortcut icon" href="/static/assets/images/logo/bird_64.png">
   <link rel="stylesheet" href="https://code.cdn.mozilla.net/fonts/fira.css">
+
+  {% compress css file vendor %} {# "vendor" css files #}
```
Comment (zzzeid) [inline src/lando/ui/jinja2/partials/layout.html:11]:
```suggestion
  {% compress css file vendor %}
```

**Review by zzzeid — COMMENTED** [review]
Remove redundant quotation marks and comments.

File: src/lando/ui/jinja2/partials/layout.html [hunk 5]
```diff
@@ -5,9 +5,41 @@
...
+
+      <link type="text/x-scss" href="/static/legacy/css/pages/StackPage.scss" rel="stylesheet" media="screen">
+      <link type="text/x-scss" href="/static/legacy/css/pages/RevisionPage.scss" rel="stylesheet" media="screen">
+      <link type="text/x-scss" href="/static/legacy/css/pages/TreestatusPage.scss" rel="stylesheet" media="screen">
+  {% endcompress %}
+
+  {% compress js file vendor %}
+      <script src="/static/legacy/vendor/jquery-3.2.1.min.js" type="text/javascript" charset="utf-8"></script>
+  {% endcompress %}
+
```
Comment (zzzeid) [inline src/lando/ui/jinja2/partials/layout.html:31]:
```suggestion
```

**Review by cgsheeh — APPROVED** [review]
Can you give a quick ELI5 for what the `{% compress js file vendor %}` directive is doing?

File: src/lando/ui/jinja2/partials/layout.html [hunk 6]
```diff
@@ -5,9 +5,40 @@
...
+      <script src="/static/legacy/js/main.js" type="text/javascript" charset="utf-8"></script>
+      <script src="/static/legacy/js/utils/formatTime.js" type="text/javascript" charset="utf-8"></script>
+
+      <script src="/static/legacy/js/components/LandingPreview.js" type="text/javascript" charset="utf-8"></script>
+      <script src="/static/legacy/js/components/Navbar.js" type="text/javascript" charset="utf-8"></script>
+      <script src="/static/legacy/js/components/Timeline.js" type="text/javascript" charset="utf-8"></script>
+      <script src="/static/legacy/js/components/FlashMessages.js" type="text/javascript" charset="utf-8"></script>
+      <script src="/static/legacy/js/components/RequestSubmitted.js" type="text/javascript" charset="utf-8"></script>
+      <script src="/static/legacy/js/components/Treestatus.js" type="text/javascript" charset="utf-8"></script>
+  {% endcompress %}
```
Comment (zzzeid) [inline src/lando/ui/jinja2/partials/layout.html:41]: For illustration, these turn into something like this upon deployments:
```html
<link rel="stylesheet" href="/static/CACHE/css/vendor.bf3fedd9e4c5.css" type="text/css" media="screen">
<link rel="stylesheet" href="/static/CACHE/css/main.7c5b759a7837.css" type="text/css" media="screen">
<script src="/static/CACHE/js/vendor.dccb4e1f0898.js"></script>
<script src="/static/CACHE/js/main.06c39d29b4f8.js"></script>
```

**Comment by zzzeid** [comment]
> Can you give a quick ELI5 for what the `{% compress js file vendor %}` directive is doing?

Added a quick description here: https://github.com/mozilla-conduit/lando/pull/73#pullrequestreview-2088733326. It basically replaces the somewhat manual asset building and linking functionality we had in Lando UI with a more automated/streamlined one that supports caching and cache invalidation.

Previously, we built assets into target files that were defined [here](https://github.com/mozilla-conduit/lando-ui/blob/main/landoui/assets_src/assets.yml), and then linked to them in the code where needed. This basically combines these two steps, and adds that additional caching/cache busting functionality.

**Comment by cgsheeh** [comment]
> > Can you give a quick ELI5 for what the `{% compress js file vendor %}` directive is doing?
> 
> Added a quick description here: [#73 (review)](https://github.com/mozilla-conduit/lando/pull/73#pullrequestreview-2088733326). It basically replaces the somewhat manual asset building and linking functionality we had in Lando UI with a more automated/streamlined one that supports caching and cache invalidation.
> 
> Previously, we built assets into target files that were defined [here](https://github.com/mozilla-conduit/lando-ui/blob/main/landoui/assets_src/assets.yml), and then linked to them in the code where needed. This basically combines these two steps, and adds that additional caching/cache busting functionality.

Thank you, that makes sense to me. :+1:
